cmd/hercules/main.go

```diff
@@ -0,0 +1,87 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"gopkg.in/src-d/go-git.v4"
+	"gopkg.in/src-d/hercules.v1"
+)
+
+func main() {
+	if len(os.Args) == 1 || len(os.Args) > 3 {
+		fmt.Fprint(os.Stderr,
+			         "Usage: hercules <path to repo or URL> [<disk cache path>]\n")
+		os.Exit(1)
+	}
+  uri := os.Args[1]
+	var repository *git.Repository
+	var err error
+	if strings.Contains(uri, "://") {
+		if len(os.Args) > 2 {
+			repository, err = git.NewFilesystemRepository(os.Args[2])
+			if err != nil {
+				panic(err)
+			}
+		} else {
+			repository = git.NewMemoryRepository()
+		}
+		err = repository.Clone(&git.CloneOptions{
+		  URL: uri,
+	  })
+	} else {
+		if uri[len(uri) - 1] == os.PathSeparator {
+			uri = uri[:len(uri) - 1]
+		}
+		if !strings.HasSuffix(uri, ".git") {
+			uri += string(os.PathSeparator) + ".git"
+		}
+		repository, err = git.NewFilesystemRepository(uri)
+	}
+	if err != nil {
+		panic(err)
+	}
+	// core logic
+	analyser := hercules.Analyser{}
+	analyser.Initialize(repository)
+	analyser.SetOnProgress(func(commit int) {
+		fmt.Fprintf(os.Stderr, "%d\r", commit)
+	})
+	statuses, last := analyser.Analyse()
+	fmt.Fprint(os.Stderr, "        \r")
+	if len(statuses) == 0 {
+		return
+	}
+	// determine the maximum length of each value
+	maxnum := int64(0)
+	for _, status := range statuses {
+		week := int64(0)
+		for i := 0; i < last; i++ {
+			val, _ := status[i]
+			week += val
+			if i % 7 == 6 {
+				if week > maxnum {
+					maxnum = week;
+				}
+				week = int64(0)
+			}
+		}
+	}
+	maxstr := strconv.FormatInt(maxnum, 10)
```

Comment:
a more idiomatic way of doing this would be using Explicit argument indexes:
```
n := 12
maxnum := 1234
width := len(fmt.Sprint(maxnum))
fmt.Printf("%[1]*[2]d", width, n)
```
will print `[space][space]12`.